sdk/aqueduct/api_client.py

```diff
@@ -107,14 +107,17 @@ class APIClient:
     NODE_POSITION_ROUTE = "/api/positioning"
     EXPORT_FUNCTION_ROUTE = "/api/function/%s/export"
 
-    def __init__(self, api_key: str, aqueduct_address: str):
+    configured = False
+
+    def configure(self, api_key: str, aqueduct_address: str) -> None:
         self.api_key = api_key
         self.aqueduct_address = aqueduct_address
 
         # Clean URL
         if self.aqueduct_address.endswith("/"):
             self.aqueduct_address = self.aqueduct_address[:-1]
 
+        self.configured = True
```

Comment:
Should configured be set at the end of this function? If we fail to validate the api_client, should we allow it to be "configured"?

We can get rid of the logic below, since that is actually solved by the constructor not taking in any arguments.
```
# If a dummy client is initialized, don't perform validation.
        if self.api_key == "" and self.aqueduct_address == "":
```